PowerPointLabs/PowerPointLabs/ActionFramework/Action/CropToSameActionHandler.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Drawing;
+using System.IO;
+using Microsoft.Office.Interop.PowerPoint;
+using PowerPointLabs.ActionFramework.Common.Attribute;
+using PowerPointLabs.ActionFramework.Common.Extension;
+using PowerPointLabs.ActionFramework.Common.Interface;
+using PowerPointLabs.CropLab;
+using Office = Microsoft.Office.Core;
+
+namespace PowerPointLabs.ActionFramework.Action
+{
+    [ExportActionRibbonId("CropToSameButton")]
+    class CropToSameActionHandler : CropLabActionHandler
+    {
+        private static readonly string ShapePicture = Path.GetTempPath() + @"\shape.png";
+        private static readonly string FeatureName = "Crop To Same Dimensions";
+
+        protected override void ExecuteAction(string ribbonId)
+        {
+            CropLabErrorHandler errorHandler = CropLabErrorHandler.InitializeErrorHandler(CropLabUIControl.GetSharedInstance());
+            if (!VerifyIsSelectionValid(this.GetCurrentSelection()))
+            {
+                HandleInvalidSelectionError(CropLabErrorHandler.ErrorCodeSelectionIsInvalid, FeatureName, CropLabErrorHandler.SelectionTypePicture, 2, errorHandler);
+                return;
+            }
+            ShapeRange shapeRange = this.GetCurrentSelection().ShapeRange;
+            if (shapeRange.Count < 2)
+            {
+                HandleInvalidSelectionError(CropLabErrorHandler.ErrorCodeSelectionIsInvalid, FeatureName, CropLabErrorHandler.SelectionTypePicture, 2, errorHandler);
+                return;
+            }
+            if (!IsPictureForSelection(shapeRange))
+            {
+                HandleErrorCodeIfRequired(CropLabErrorHandler.ErrorCodeSelectionMustBePicture, FeatureName, errorHandler);
+                return;
+            }
+            var refShape = shapeRange[1];
+            float refScaleWidth = PowerPointLabs.Utils.Graphics.GetScaleWidth(refShape);
+            float refScaleHeight = PowerPointLabs.Utils.Graphics.GetScaleHeight(refShape);
+            //MessageBox.Show(shapes[1].PictureFormat.CropTop.ToString() + " " + shapes[1].PictureFormat.CropBottom.ToString() + " " + shapes[1].Height.ToString() + " " + );
+            //refShape.ScaleHeight(0.5F, Microsoft.Office.Core.MsoTriState.msoFalse);
+            float epsilon = 0.001F;
```

Comment:
put below logic into a logic class, e.g. maybe CropToSame, instead of mixing it into the action